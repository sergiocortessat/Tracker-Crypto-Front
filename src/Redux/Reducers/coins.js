const initialState = {
  coins: [],
};

const coinReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_COINS':
      return {
        ...state,
        coins: action.payload,
      };
    default:
      return state;
  }
};

export default coinReducer;
