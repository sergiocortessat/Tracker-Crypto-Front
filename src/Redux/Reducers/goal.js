const initialState = {
  goals: [],
};

const goalReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_GOALS':
      return {
        ...state,
        goal: action.payload,
      };
    default:
      return state;
  }
};

export default goalReducer;
