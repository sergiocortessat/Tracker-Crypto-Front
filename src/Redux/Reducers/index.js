import { combineReducers } from 'redux';
import Coins from './coins';
import User from './user';

const reducers = combineReducers({
  coins: Coins,
  user: User,
});

export default reducers;
