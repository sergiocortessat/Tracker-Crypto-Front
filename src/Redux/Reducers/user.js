const initialState = {
  user: [],
};

const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'CURRENT_USER':
      return {
        ...state,
        user: action.payload,
      };
    default:
      return state;
  }
};

export default userReducer;
