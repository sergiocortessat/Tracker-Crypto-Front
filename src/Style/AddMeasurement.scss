.add-measurement {
  display: flex;
  flex-direction: column;
  align-items: center;

  h1 {
    margin-top: 50px;
  }

  form {
    margin-top: 50px;
    text-align: center;
    background-color: rgba(108, 122, 137, 0.7);
    width: 280px;
    border-radius: 15px;

    input {
      width: 120px;
      margin-top: 20px;
    }

    .main-label {
      display: flex;
      flex-direction: column;
      width: 250px;
      margin: 0 auto 20px auto;
      height: 200px;
      align-items: center;

      span {
        margin-top: 15px;
        margin-bottom: 30px;
      }

      select {
        width: 120px;
      }

      .second-label {
        input {
          width: 120px;
        }
      }
    }
  }
}
