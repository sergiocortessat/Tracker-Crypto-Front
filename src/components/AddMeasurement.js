/* eslint-disable no-unused-vars */
import { useAuth0 } from '@auth0/auth0-react';
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getUsers, postMeasurements } from '../API/API';
import { updateUser } from '../Redux/Actions';

const AddMeasurement = () => {
  const { coins } = useSelector((state) => state.coins);
  const [coin, setCoin] = useState(1);
  const dispatch = useDispatch();
  const currentUser = useSelector((state) => state.user);
  const { user, isAuthenticated, getAccessTokenSilently } = useAuth0();
  const [unit, setUnit] = useState();
  const [limit, setLimit] = useState(1);

  useEffect(() => {
    getUsers().then((res) => {
      const filtered = res.filter((a) => a.sub === user.sub);
      // const goals = filtered[0].coins.filter((a) => a.id === coin)[0];
      // const { goal } = goals.goals.filter((g) => g.sub === user.sub)[0];
      // setLimit(goal);
      dispatch(updateUser(filtered[0]));
    });
  }, []);

  // useEffect(() => {
  //   setTimeout(() => {
  //     const tempGoal = currentUser.length < 1 ? currentUser.user.coins
  //       .filter((c) => c.id === coin)[0] : 1;
  //     const { goal } = tempGoal.length < 1 ? tempGoal.filter((g) => g.sub === user.sub)[0] : 1;
  //     setLimit(goal);
  //   }, 1000);
  // });

  const handleSubmit = (e) => {
    e.preventDefault();
    const tempGoal = currentUser.user.coins.filter((c) => c.id === coin)[0].goals;
    const goalID = tempGoal.filter((g) => g.sub === user.sub)[0].id;
    const { goal } = tempGoal.filter((g) => g.sub === user.sub);
    const userID = currentUser.user.id;
    getAccessTokenSilently()
      .then((accessToken) => {
        postMeasurements({ user_id: userID, goal_id: goalID, unit }, accessToken);
      });
  };
  console.log(limit);
  return (
  // create a form with list items and
    <form action="/action_page.php">
      <label htmlFor="coins">
        Choose a Crypto:
        <select id="coins" name="coins">
          {coins && coins.map((coin) => (
            <option
              key={coin.id}
              value={coin.name}
              onClick={() => setCoin(coin.id)}
            >
              {coin.name}
            </option>
          ))}
          )
        </select>
        {/* <input
          type="text"
          placeholder="Units"
          onChange={(event) => setUnit(event.currentTarget.value)}
          value={unit}
        /> */}
        <label htmlFor="quantity">
          <input placeholder="Units" value={unit} type="number" id={coin} step={0.01} min="0" max={limit} onChange={(e) => setUnit(e.target.value)} />

        </label>
        <input type="submit" value="Submit" id={coin} onClick={(e) => handleSubmit(e)} />
      </label>
    </form>

  );
};

export default AddMeasurement;
