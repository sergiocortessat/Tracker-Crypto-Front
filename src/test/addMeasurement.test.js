import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import Reducers from '../Redux/Reducers';
import AddMeasurement from '../components/AddMeasurement';

const store = createStore(Reducers);

it('renders correctly', () => {
  const addMeasurement = renderer
    .create(
      <Provider store={store}>
        <AddMeasurement />
      </Provider>,
    )
    .toJSON();
  expect(addMeasurement).toMatchSnapshot();
});
