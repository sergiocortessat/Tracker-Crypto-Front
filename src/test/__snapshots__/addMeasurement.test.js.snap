// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="add-measurement"
>
  <h1>
    Add a Measurement
  </h1>
  <form>
    <label
      className="main-label"
      htmlFor="coins"
    >
      <span>
        Choose a Crypto:
      </span>
      <select
        id="coins"
        name="coins"
        onChange={[Function]}
      />
      <label
        className="second-label"
        htmlFor="quantity"
      >
        <input
          id={1}
          max={10}
          min="0"
          onChange={[Function]}
          placeholder="Units"
          step={0.01}
          type="number"
          value=""
        />
      </label>
      <input
        id={1}
        onClick={[Function]}
        type="submit"
        value="Submit"
      />
    </label>
  </form>
</div>
`;
